pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -1097,19 +1097,30 @@ func (a *Actuator) remediateIfNeeded(ctx context.Context, machine *machinev1beta
 	}
 
 	if _, needsRemediation := machine.Annotations[externalRemediationAnnotation]; !needsRemediation {
-		return nil
-	}
-
-	node, err := a.getNodeByMachine(ctx, machine)
+		// It can happen that an unhealthy machine with external remediation annotation got healthy, and the external
+		// remediation annotation was removed by the MHC controller, before remediation finished or even started
+		// (e.g. the metal3 pod is not running, but machine was rebooted manually).
+		// In this case stop remediation, if possible.
+
+		// If bmh isn't requested to power off, and the machine isn't marked as powered off for remediation,
+		// there is no ongoing remediation, and there is nothing to do
+		powerOffRequestExists := hasPowerOffRequestAnnotation(baremetalhost)
+		_, poweredOffForRemediationExists := machine.Annotations[poweredOffForRemediation]
+		if !powerOffRequestExists && !poweredOffForRemediationExists {
+			return nil
+		}
 
-	if err != nil {
-		if !errors.IsNotFound(err) {
-			log.Printf("Failed to get Node from Machine %s: %s", machine.Name, err.Error())
-			return err
+		// If bmh was requested to power off, but the machine isn't marked as powered off for remediation yet:
+		// remediation just started and it's fine to stop it by removing the power off request.
+		// That way the bmh won't be powered off at all, or immediately powered on again.
+		if powerOffRequestExists && !poweredOffForRemediationExists {
```

Comment:
I'm a bit confused.
What I understand from the code is:
1.  Power Off was requested. (`powerOffRequestExists == true`)
2. The Origin of the Power Off request was not due to remediation. (`poweredOffForRemediationExists == false && needsRemediation == false`). 

In this case I'd expect the Power Off to continue as planned.
What am I missing here ?

Reply:
> The Origin of the Power Off request was not due to remediation

no, the annotation indicating this just wasn't set yet. So I know we are at a very early stage of remediation and it's safe to stop it